### DISCUSSION TITLE
can I limit access to one ip address?

### DISCUSSION THREAD

**Discussion by mmoalem** [spencerwooo/onedrive-vercel-index#787] [Q&A]
HI - I would like to use this script to bridge between my website and onedrive. My website has a mysql database of music tracks and a search page that returns a list of tracks matching the search keyword. on each returned track there is an audio player that can play the track and a download button to download the tracks and they rely on URLs for the tracks. My web server is hosted on OCI with not enough storage to host all the music and I have E5 5tb onedrive space but onedrive does not provide direct URLs.
This script seems to offer direct URL links but I want it to be hidden, only accessed by my website... any ideas if its possible to restrict access?

**Comment by spencerwooo** [comment]
> This script seems to offer direct URL links...

This repo only provides URLs that **redirect** to the direct link provided by the OneDrive API. Hence, you would still access your resource through the direct link in OneDrive - which is not IP-protectable.

If you are using your OneDrive as an AWS S3-like object storage - which from your description seems like so - I wouldn't bother restricting access like this. You would still want visitors of your website to play audio resources hosted on your OneDrive, which would ultimately provide users access to your OneDrive direct links.

If you insist, I would suggest hosting your website on Cloudflare and using their firewall for IP restrictions?

**Comment by mmoalem** [comment]
Hi and thank you very much for the reply! I guess what I would like to avoid is having the vercel app interface available to the public. I was after a way to index onedrive and generate direct urls so I offer my site users who has password protected access the abilty to stream and download the music - this script seems to do it but I rather it was not 'published' online

can I run the vercel front end on my own server that I can hide?

I am just trying out an idea that is not making any money so rather keep to using free resources for now

**Comment by spencerwooo** [comment]
> I guess what I would like to avoid is having the vercel app interface available to the public.

Then I guess you won't need to use this project at all. You could simply use what's under `pages/api/`, specifically `pages/api/index.ts`, where the call to the OneDrive API happens. You can use this bit of logic and make your own calls to the OneDrive API for direct link queries - which should be easier to implement on your server.

**Reply by mmoalem** [reply]
thanks again - appreciate your support. unfortunately I have no coding skills so what you suggested is way beyond my understanding... I am ok at following instructions to use script like yours but no idea how they work or how to modify them so looking at index.ts it looks like martian to me... Will keep looking for a solution... thanks again for trying to help
